src/main.c

```diff
@@ -101,19 +101,71 @@ static void onGlfwError(int code, const char* description) {
   lovrThrow(description);
 }
 
+typedef enum { // What is the argument parser doing?
+  argparse_exe, // Interpreter
+  argparse_flags,
+  argparse_game // Game path and arguments
+} ArgParseState;
+
+typedef enum { // In state argparse_flag, what flag is being searched for?
+  argflag_none, // Not processing a flag
+  argflag_inside
+} ArgFlag;
+
 lua_State* lovrInit(lua_State* L, int argc, char** argv) {
   glfwSetErrorCallback(onGlfwError);
   lovrAssert(glfwInit(), "Error initializing GLFW");
   glfwSetTime(0);
 
-  // arg
+  // arg table
+  // Args follow the lua standard https://en.wikibooks.org/wiki/Lua_Programming/command_line_parameter
+  // * The lovr interpreter and its arguments are in successive negative indexes starting at -1.
+  //   The interpreter will always be at arg[-1] and always non-nil.
```

Comment:
I tested this script:

```lua
for i = -3, 3 do
  print(arg[i])
end
```

like this:

```sh
lua -E test.lua
```

and got

```
nil
lua
-E
test.lua
hi
nil
nil
```

So I think it's reversed and the interpreter is at the "lowest" index.  I haven't yet read on enough to see if this impacts the parsing.